src/imp/gen_map_get.rs

```diff
@@ -116,13 +116,14 @@ impl Stmt {
   }
 }
 
-type Substitutions = HashMap<Name, (Name, Box<Expr>)>;
+type Substitutions = IndexMap<Name, (Name, Box<Expr>)>;
```

Comment:
If all we need is to do an rfold, we could have Substitutions be a simple `Vec<(Name, Name, Box<Expr>)>`